network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -335,10 +364,31 @@ impl ConnectionPoolBehaviour {
     /// Tells the behaviour to stop connecting to other peers.
     /// This is useful when we are sure we have no possibility of getting a
     /// connection such as in a network outage.
-    pub fn stop_connecting(&mut self) {
+    fn stop_connecting(&mut self) {
         self.active = false;
     }
 
+    /// Closes a peer connection with a reason
+    ///
+    /// This will take actions depending on the close reason. For instance:
+    /// - Malicious peer will ban the connection.
```

Comment:
A malicious peer..

Reply:
> 

A malicious peer..
The close reason `MaliciousPeer` will cause the peer to be banned.